codec/src/main/java/io/netty/handler/codec/BatchFixedLengthFrameDecoder.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.codec;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+
+import java.util.List;
+
+/**
+ * A decoder that mostly mimics {@link FixedLengthFrameDecoder} semantics. The only difference is this decoder batches
+ * incoming {@link ByteBuf} messages and then decodes them at once, passing a big {@link ByteBuf} which can possibly
+ * contain N messages (where N >= 1) through the pipeline at once.
+ */
+public class BatchFixedLengthFrameDecoder extends BatchFrameDecoder {
+
+    final int frameLength;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param frameLength the length of the frame
+     */
+    public BatchFixedLengthFrameDecoder(int frameLength) {
+        this(0, frameLength);
+    }
+
+    /**
+     * Creates a new instance.
+     *
+     * @param startBufferSize the start buffer size which is used for batching incoming {@link ByteBuf} messages. Size
+     * of this buffer must be greater or equal to zero. If the size of this buffer is zero then the default size will be
+     * used (= 256 bytes).
+     * @param frameLength the length of the frame
+     */
+    public BatchFixedLengthFrameDecoder(int startBufferSize, int frameLength) {
+        super(startBufferSize);
+
+        if (frameLength <= 0) {
+            throw new IllegalArgumentException("frameLength must be a positive integer: " + frameLength);
+        }
+
+        this.frameLength = frameLength;
+    }
+
+    @Override
+    protected void decode(ChannelHandlerContext ctx, ByteBuf buffer, List<Object> out) {
+        final int readableBytes = buffer.readableBytes();
+
+        if (readableBytes < frameLength) {
+            return;
+        }
+
+        final int chunkLength = buffer.readableBytes() / frameLength * frameLength;
```

Comment:
Let's suppose we have 1405 bytes in the `buffer`. If we divide the number of readable bytes by the frame length (100, for example), we'll get 14 as a result. This will be the number of messages in our `buffer` variable. 

Now we can multiply the result by the frame length again in order to get the number of bytes wee need to copy. By multiplying the result by 100 we'll get 1400 bytes. Now we can simply create a new `ByteBuff`, copy 1400 bytes into it and pass it further through the pipeline. 

This new `ByteBuff` will contain 14 messages. The remainder (5 bytes) can now be moved to the beginning of our `buffer` variable, so we can continue buffering incoming data.